pkg/pool-linear/contracts/euler/EulerLinearPoolRebalancer.sol

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-interfaces/contracts/pool-utils/ILastCreatedPoolFactory.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/SafeERC20.sol";
+import "./interfaces/IEulerTokenMinimal.sol";
+
+import "../LinearPoolRebalancer.sol";
+
+contract EulerLinearPoolRebalancer is LinearPoolRebalancer {
+    using SafeERC20 for IERC20;
+
+    // These Rebalancers can only be deployed from a factory to work around a circular dependency: the Pool must know
+    // the address of the Rebalancer in order to register it, and the Rebalancer must know the address of the Pool
+    // during construction.
+    constructor(IVault vault, IBalancerQueries queries)
+        LinearPoolRebalancer(ILinearPool(ILastCreatedPoolFactory(msg.sender).getLastCreatedPool()), vault, queries)
+    {
+        // solhint-disable-previous-line no-empty-blocks
+    }
+
+    function _wrapTokens(uint256 amount) internal override {
+        // No referral code, depositing from underlying (i.e. DAI, USDC, etc. instead of aDAI or aUSDC). Before we can
+        // deposit however, we need to approve the wrapper in the underlying token.
+
+        // approve Euler Mainnet
+        _mainToken.safeApprove(0x27182842E098f60e3D576794A5bFFb0777E025d3, amount);
```

Comment:
address(_wrappedToken) doesn't work in here?

Reply:
@joaobrunoah it is actually not the wrapped token that holds the balance but the Euler Mainnet contract. 
see: https://docs.euler.finance/developers/getting-started/integration-guide. 

`// Approve the main euler contract to pull your tokens:
IERC20(underlying).approve(EULER_MAINNET, type(uint).max);`

Reply:
@mkflow27 Yes, we should avoid hard-coding any network-specific data. The Balancer Vault, for example, accepts a WETH address in its constructor because WETH (or more generally a native token wrapper) has different addresses on each network. I would do something like this:

1. Add an argument for the Euler protocol address to both the Rebalancer's and the Factory's constructors.
2. In the Factory's constructor, copy the incoming Euler protocol address to storage.
3. In the Factory's `create` function, read the stored address and pass it into the Rebalancer's constructor.
4. In the Rebalancer's constructor, copy the incoming Euler protocol address to storage. Read it when needed, as in the line above.

This architecture allows each Factory to store the canonical Euler protocol address that will be automatically trickled down to all Rebalancers from that Factory.